src/components/AssetPage/index.js

```diff
@@ -54,9 +59,17 @@ const mapStateToProps = (state, props) => ({
   counterValue: counterValueCurrencySelector(state),
   allAccounts: accountsSelector(state),
   countervalueFirst: countervalueFirstSelector(state),
-  accounts: flattenSortAccountsEnforceHideEmptyTokenSelector(state).filter(
-    a => getAccountCurrency(a) === findCurrencyByTicker(props.match.params.assetTicker),
-  ),
+  accounts: flattenSortAccountsEnforceHideEmptyTokenSelector(state).filter(a => {
+    let currency
+    const { assetId, assetType } = props.match.params
+    if (assetType === 'TokenCurrency') {
+      currency = findTokenById(assetId)
+    } else {
+      currency = findCryptoCurrencyById(assetId)
+    }
+    const accountCurrency = getAccountCurrency(a)
+    return accountCurrency === currency
```

Comment:
I feel this code can be replaced by  `filter(a => getAccountCurrency(a).id === props.match.params.assetId)`